### PR TITLE
Fix segment double-counting and consolidate _app code

### PR DESCRIPTION
**Pull request by jamieparkinson**
Resolves https://github.com/wellcomecollection/wellcomecollection.org/issues/8476

## The bug and the fix
We were double-counting in production builds because we were [calling](https://github.com/wellcomecollection/wellcomecollection.org/blob/e921b7375e577882f014c4748797260672608e08/common/views/pages/_app.tsx#L347-L352) the `trackPageview` method on client-side renders: a side-effect in the render phase in React is never a good idea!

`useEffect` with an empty dependencies array is the right tool for the job of fixing this, but as #8476 points out, it's not the full picture: it'll run once per client-side render, but when we navigate within a Next.js app, our `_app` doesn't re-render, and so the method won't get called.

Luckily, the Next.js [router](https://nextjs.org/docs/api-reference/next/router) provides us with some event listeners for page transitions, and we can use these to trigger additional pageview events: this PR implements a general purpose `useOnPageLoad` hook for triggering side effects once per page (whether that's on the first load from the server, or on future transitions), and uses that hook to fix this bug ✨ .

## The refactor
`_app.tsx` contained _a lot_ of separate concerns related to "global" site functionality. That's what it's for, but it was hard to follow these concerns, and I've split them out into a series of hooks and other utilities which now live in `common/services/app`. There were some types that I cleaned up in the process of doing this.

The segment code (`common/services/conversion/track.ts`) is still a little confusing in my opinion, especially given the overloaded use of the word "track" for both GA and Segment. It's good enough for now though, and regardless it shouldn't be touched in this already wide-ranging PR.

### CODE REVIEW

File: common/utils/ga.ts [hunk 1]
```diff
@@ -1,14 +1,8 @@
...
-  category: string;
-  action: string;
-  label?: string;
-  value?: number;
-  nonInteraction?: boolean;
-  transport?: 'beacon';
-};
+import ReactGA, { EventArgs as GaEvent } from 'react-ga';
 
+// TODO this wrapper function is redundant but pervasive; consider getting rid of it
```
Comment (alexwlchan) [inline common/utils/ga.ts:3]: get rid of as in replace with an inline call to `ReactGA.event(gaEvent)`?

File: common/views/pages/_document.tsx [hunk 2]
```diff
@@ -60,20 +64,8 @@ class WecoDoc extends Document {
...
-            dangerouslySetInnerHTML={{
-              __html: `
-                  window.dataLayer = window.dataLayer || [];
-                  function gtag(){dataLayer.push(arguments);}
-                  gtag('js', new Date());`,
-            }}
-          />
-
+          <GoogleAnalyticsV4 />
+          <GoogleAnalyticsUA />
```
Comment (alexwlchan) [inline common/views/pages/_document.tsx:68]: I have a vague memory that we don't components certain elements in the `<head>` (I think `<meta>` tags because they have to be declare at the top level; I assume that's not an issue here?

**Review by alexwlchan — APPROVED** [review]
Breaking this into nicely named hooks in separate files seems sensible and makes this easier to follow, a couple of non-blocking Qs but mostly looks good. 📈 👍

File: common/utils/ga.ts (hunk 1 above)
Comment (jamieparkinson) [reply common/utils/ga.ts:3]: I started doing this but its usage is _extensive_ and since we will be getting rid of UA I thought it probably unnecessary

File: common/views/pages/_document.tsx (hunk 2 above)
Comment (jamieparkinson) [reply common/views/pages/_document.tsx:68]: It seems to work fine; it does seem like the way the nextjs `Head` component works and what it tolerates might've changed/expanded, respectively, over the years

File: common/utils/ga.ts (hunk 1 above)
Comment (jamieparkinson) [reply common/utils/ga.ts:3]: Oh sorry I misread this comment - yes, correct
